src/main/java/com/verily/genomewarp/utils/GvcfToVcfAndBed.java

```diff
@@ -19,92 +19,145 @@
 import java.util.logging.Logger;
 
 /**
- * Helper class for extracting a canonical VCF and a BED file from a gVCF file
+ * Helper class for extracting a canonical VCF and a BED file from a gVCF file.
  */
 public class GvcfToVcfAndBed {
-
-  private static final Logger logger = Logger.getLogger(VcfToVariant.class.getName());
-
+  // class constants
+  private static final Logger logger = Logger.getLogger(GvcfToVcfAndBed.class.getName());
+  
+  private static final int MIN_VCF_FIELDS = 8;
+  private static final int START_FIELD_INDEX = 1;
+  private static final int END_FIELD_INDEX = 3;
+  private static final int ALT_BASES_FIELD_INDEX = 4;
+  private static final int FILTER_FIELD_INDEX = 6;
+  private static final int FORMAT_FIELD_INDEX = 7;
+  private static final String UNKNOWN_VALUE = ".";
+  private static final String GVCF_ALT_ALLELE = "<*>";
+  private static final String HEADER_START_STR = "#";
+  private static final String FIELD_SEP_REGEX = "\\t";
+  private static final String PASS_FILTER_VALUE = "PASS";
+  private static final String FORMAT_FIELD_SEP_REGEX = ";";
+  private static final String FORMAT_END_KEY = "END";
+  private static final int CHROM_FIELD_INDEX = 0;
+  private static final String BED_FIELD_SEP_STR = "\t";
+  
   /**
    * Solve Issue #2 - Extract variant VCF and BED from (tentative) gVCF file
    * 
-   * @param rawQueryVcf user-provided (g?)VCF path
+   * @param rawQueryGvcf user-provided gVCF path
    * @param vcfFromGvcf intermediate canonical VCF path
    * @param bedFromGvcf intermediate BED path
-   * @return true if query VCF is a gVCF, false otherwise
+   * @return true if query gVCF was parsed successfully, false otherwise
    */
-  public static boolean saveVcfAndBedFromGvcf(String rawQueryVcf, String vcfFromGvcf,
-      String bedFromGvcf) throws IOException {
-    PrintWriter outBed = new PrintWriter(Files.newBufferedWriter(Paths.get(bedFromGvcf), UTF_8));
-    PrintWriter outVcf = new PrintWriter(Files.newBufferedWriter(Paths.get(vcfFromGvcf), UTF_8));
-    BufferedReader br = Files.newBufferedReader(Paths.get(rawQueryVcf), UTF_8);
+  public static boolean saveVcfAndBedFromGvcf(String rawQueryGvcf, String vcfFromGvcf,
+      String bedFromGvcf) {
     String line;
-    boolean isGvcf = false;
+    boolean isSuccessfulGvcf = false;
     Long lineNum = 0L;
-    while ((line = br.readLine()) != null) {
-      // parse line and write it to VCF or BED or none
-      lineNum++;
-      if (line.length() > 0) {
-        if (line.startsWith("#")) {
-          // header
-          outVcf.println(line);
-        } else {
-          // not a header - choose destination according to columns
-          String[] fields = line.split("\\t");
-          if (fields.length < 8) {
-            throw new IOException("input VCF file has less than 8 columns");
-          }
-          String f5 = fields[4];
-          if (!".".equals(f5) && !"<*>".equals(f5)) {
-            // variant VCF file
+    try {
```

Comment:
I think the nesting of trys can be avoided and increase clarity.

Path bedFromGvcfPath;
...
try {
  bedFromGvcfPath = Paths.get(bedFromGvcf);
  ...
} catch (Exception ex) {
  logger.log(Level.SEVERE, "Error creating paths: " + ex.toString());
  return false;
}

try (PrintWriter ...) {
  ...
}

Reply:
(Or just put the Paths.get calls within the try-with-resources block so they get caught there)

Reply:
I'll use the first option (like I did originally, don't know why I changed =) ), since I need the intermediate files paths available outside the try scope when I remove them on error.